# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/CodecConfig.h
# Commit: 09db50c8f370
# Full Hash: 09db50c8f370b69f186f3e6b7ce6154a984b40b4
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 9
# Description:
#   Bug 1654112 - Pass codec configs by value/reference instead of pointers. r=ng
#   
#   This patch mainly sets the stage for using codec configs with StateMirroring.
#   It also makes the code a bit simpler, and removes the need to handle the nullptr
#   case.
# ==============================================================================

diff -r d1552c620c11 -r 09db50c8f370 dom/media/webrtc/libwebrtcglue/CodecConfig.h
--- a/dom/media/webrtc/libwebrtcglue/CodecConfig.h	Wed Aug 11 11:33:21 2021 +0200
+++ b/dom/media/webrtc/libwebrtcglue/CodecConfig.h	Wed Apr 14 17:36:40 2021 +0200
@@ -53,6 +53,20 @@
         mMaxAverageBitrate(0),
         mMaxPlaybackRate(0),
         mCbrEnabled(false) {}
+
+  bool operator==(const AudioCodecConfig& aOther) const {
+    return mType == aOther.mType && mName == aOther.mName &&
+           mFreq == aOther.mFreq && mChannels == aOther.mChannels &&
+           mFECEnabled == aOther.mFECEnabled &&
+           mDtmfEnabled == aOther.mDtmfEnabled &&
+           mFrameSizeMs == aOther.mFrameSizeMs &&
+           mMaxFrameSizeMs == aOther.mMaxFrameSizeMs &&
+           mMinFrameSizeMs == aOther.mMinFrameSizeMs &&
+           mDTXEnabled == aOther.mDTXEnabled &&
+           mMaxAverageBitrate == aOther.mMaxAverageBitrate &&
+           mMaxPlaybackRate == aOther.mMaxPlaybackRate &&
+           mCbrEnabled == aOther.mCbrEnabled;
+  }
 };
 
 /*
@@ -68,6 +82,14 @@
   int packetization_mode;
   int profile_level_id;
   int tias_bw;
+
+  bool operator==(const VideoCodecConfigH264& aOther) const {
+    return strncmp(sprop_parameter_sets, aOther.sprop_parameter_sets,
+                   MAX_SPROP_LEN) == 0 &&
+           packetization_mode == aOther.packetization_mode &&
+           profile_level_id == aOther.profile_level_id &&
+           tias_bw == aOther.tias_bw;
+  }
 };
 
 // class so the std::strings can get freed more easily/reliably
@@ -112,25 +134,22 @@
   // TODO: add external negotiated SPS/PPS
 
   bool operator==(const VideoCodecConfig& aRhs) const {
-    if (mType != aRhs.mType || mName != aRhs.mName ||
-        mAckFbTypes != aRhs.mAckFbTypes || mNackFbTypes != aRhs.mNackFbTypes ||
-        mCcmFbTypes != aRhs.mCcmFbTypes || mRembFbSet != aRhs.mRembFbSet ||
-        mFECFbSet != aRhs.mFECFbSet ||
-        mTransportCCFbSet != aRhs.mTransportCCFbSet ||
-        mULPFECPayloadType != aRhs.mULPFECPayloadType ||
-        mREDPayloadType != aRhs.mREDPayloadType ||
-        mREDRTXPayloadType != aRhs.mREDRTXPayloadType ||
-        mRTXPayloadType != aRhs.mRTXPayloadType || mTias != aRhs.mTias ||
-        !(mEncodingConstraints == aRhs.mEncodingConstraints) ||
-        !(mEncodings == aRhs.mEncodings) ||
-        mSpropParameterSets != aRhs.mSpropParameterSets ||
-        mProfile != aRhs.mProfile || mConstraints != aRhs.mConstraints ||
-        mLevel != aRhs.mLevel ||
-        mPacketizationMode != aRhs.mPacketizationMode) {
-      return false;
-    }
-
-    return true;
+    return mType == aRhs.mType && mName == aRhs.mName &&
+           mAckFbTypes == aRhs.mAckFbTypes &&
+           mNackFbTypes == aRhs.mNackFbTypes &&
+           mCcmFbTypes == aRhs.mCcmFbTypes && mRembFbSet == aRhs.mRembFbSet &&
+           mFECFbSet == aRhs.mFECFbSet &&
+           mTransportCCFbSet == aRhs.mTransportCCFbSet &&
+           mULPFECPayloadType == aRhs.mULPFECPayloadType &&
+           mREDPayloadType == aRhs.mREDPayloadType &&
+           mREDRTXPayloadType == aRhs.mREDRTXPayloadType &&
+           mRTXPayloadType == aRhs.mRTXPayloadType && mTias == aRhs.mTias &&
+           mEncodingConstraints == aRhs.mEncodingConstraints &&
+           mEncodings == aRhs.mEncodings &&
+           mSpropParameterSets == aRhs.mSpropParameterSets &&
+           mProfile == aRhs.mProfile && mConstraints == aRhs.mConstraints &&
+           mLevel == aRhs.mLevel &&
+           mPacketizationMode == aRhs.mPacketizationMode;
   }
 
   VideoCodecConfig(int type, std::string name,